Add sanitization for sensitive data in aws-serverless-mcp-server

greyshi opened the pull request:
## Add Response Sanitization for Sensitive Data in CloudWatch Logs

### Description

This PR adds data sanitization capabilities to the AWS Serverless MCP Server to prevent leaking sensitive information in CloudWatch logs and Lambda function outputs.

### What's Changed

- Created a new `ResponseSanitizer` class in `utils/security.py` that redacts sensitive patterns including:
  - AWS access keys and secret keys
  - Passwords in various formats
  - AWS account IDs (12-digit numbers)
  - API keys (including x-api-key headers)

- Integrated sanitization into two tools which are capable of outputting sensitive data:
  - `sam_logs` - Sanitizes CloudWatch logs output when fetching Lambda logs
  - `sam_local_invoke` - Sanitizes both function output and execution logs

- Added test coverage for this feature

### Implementation Details

The sanitization only applies when the `--allow-sensitive-data` flag is enabled, since otherwise the data would not be returned anyways. The implementation uses regex patterns to identify and redact sensitive information while preserving the overall structure and readability of logs.

### Testing

All tests pass successfully:

```
$ python -m unittest tests.test_response_sanitization -v
test_sanitize_cloudwatch_logs ... ok
test_sanitize_dict ... ok
test_sanitize_empty_structures ... ok
test_sanitize_lambda_invocation_response ... ok
test_sanitize_list ... ok
test_sanitize_non_string_types ... ok
test_sanitize_string ... ok

----------------------------------------------------------------------
Ran 7 tests in 0.001s

OK
```

### Example Output

Before sanitization:
```
2024-01-15T10:30:01.000Z [INFO] AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE
2024-01-15T10:30:01.000Z [INFO] AWS_SECRET_ACCESS_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
2024-01-15T10:30:02.000Z [INFO] Processing account 123456789012
2024-01-15T10:30:02.000Z [INFO] Database password=supersecretdbpass
```

After sanitization:
```
2024-01-15T10:30:01.000Z [INFO] AWS_ACCESS_KEY_ID=[REDACTED AWS_ACCESS_KEY]
2024-01-15T10:30:01.000Z [INFO] AWS_SECRET_ACCESS_KEY=[REDACTED AWS_SECRET_KEY]
2024-01-15T10:30:02.000Z [INFO] Processing account [REDACTED AWS_ACCOUNT_ID]
2024-01-15T10:30:02.000Z [INFO] Database [REDACTED PASSWORD]
```

### Files Changed

- `awslabs/aws_serverless_mcp_server/utils/security.py` - New file containing ResponseSanitizer
- `awslabs/aws_serverless_mcp_server/tools/sam/sam_logs.py` - Added sanitization to log output
- `awslabs/aws_serverless_mcp_server/tools/sam/sam_local_invoke.py` - Added sanitization to function output
- `tests/test_response_sanitization.py` - New test file for ResponseSanitizer
- `tests/test_sam_tools_sanitization.py` - New test file for tool integration

scottschreckengaust commented:
Thank you for the contribution. Please run pre-commit and review auditing detected secrets from the developer's guide

bx9900 commented on src/aws-serverless-mcp-server/awslabs/aws_serverless_mcp_server/utils/security.py:
```diff
@@ -0,0 +1,82 @@
+        "aws_secret_key": r"\b[A-Za-z0-9/+=]{40}\b",
+        "password": r"(?i)password\s*[=:]\s*[^\s]+",
+        "private_key": r"-----BEGIN (?:RSA|DSA|EC|OPENSSH) PRIVATE KEY-----",
+        "aws_account_id": r"\b\d{12}\b",
```
Curious why sanitize account ids?

greyshi replied:
Included for consistency since the ECS sanitization is filtering it. I suppose since it's one piece of the puzzle to get access to an AWS account even though it's not one of the important ones. Don't feel strongly either way, so let me know if you think we should remove it.

bx9900 replied:
I think it's fine to keep consistent with ECS.

greyshi commented:
pre-commit had passed when I ran it locally. Will try to reproduce and fix the 3 failing checks above.

scottschreckengaust commented:
The `.secrets.baseline` has some corrections, and prior audited items that have been removed.

scottschreckengaust reviewed (changes requested):
Detect secrets update and pyright

greyshi commented:
Thanks. I was having an issue where pyright was succeeding locally with no findings. This happened because the aws-serverless-mcp-server project was missing a `pyrightconfig.json` configuration file, causing pyright to run with different settings locally vs in CI. Added that file and have addressed the issues. Will send a new commit with the fixes.
